### PR TITLE
Pytree refactor

### PR DESCRIPTION
**Pull request by billbrod**
Just opening this right now so that we can get the GLM example rendered by read the docs. More work needs to be done here.

For the example in particular:
- I'm using a dataset from DANDI, looking at head direction and 2d position.
- I think this example is too complex to introduce the pytrees and more/better text is needed here.
- Writing this made it clear that we can easily put together some simple helper functions that will make things easier, like doing the multiplication between bases and coefficients.
- When this goes into the docs, the general "here's a pytree" should probably be a separate tutorial from "here's how to use them with GLMs".
- I added some extra dependencies in pyproject.toml, but that's just so this example builds. We should think about how we want to handle this.

### CODE REVIEW

File: docs/examples/plot_glm_pytree.py [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+plt.figure()
+plt.plot(x, basis.evaluate(x))
+
+# this is the interval on which head_dir has no NaNs and spikes.count() returns values
+valid_data = nap.IntervalSet(.88, 599.91)
+head_dir = head_dir.restrict(valid_data)
+spikes = spikes.count(bin_size=.02, time_units='s').restrict(valid_data)
+
+# add extra dim for n_neurons (here, 1)
+X = nmo.pytrees.FeaturePytree(head_direction=np.expand_dims(basis.evaluate(head_dir.values), 1))
```
[the remark this answers is not in the document]
Comment (gviejo) [reply docs/examples/plot_glm_pytree.py:224]: Also why the expand_dims?

[the remark this answers is not in the document]
Comment (billbrod) [reply docs/examples/plot_glm_pytree.py:224]: as discussed, expand_dims adds an extra dimension for (the single) neurons. We should think about how to handle this, but it's not a pytree problem

**Comment by billbrod** [comment]
I addressed Guillaume's comments and added tests for pytrees. There weren't many, but I think they're sufficient. Let me know if there's anything else I should add to them.

**Comment by billbrod** [comment]
Currently, one test is failing:
- `test_fit_init_params_type`: when input is `iter` or a dict with keys `p1,p2`. The problem is that we expect `params` to be a 2-tuple (in order  to be compliant with sklearn, I thin), but this leads to some weirdness, as in these cases. We should talk through this

File: docs/examples/plot_glm_pytree.py [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+print(mapped)
+mapped['feature_1']
+# %%
+# Or functions from jax or numpy that operate on arrays:
+mapped = jax.tree_map(jnp.exp, example_pytree)
+print(mapped)
+mapped['feature_1']
+# %%
+# We can change the dimensionality of our pytree:
+mapped = jax.tree_map(lambda x: jnp.mean(x, axis=-1), example_pytree)
```
[the remark this answers is not in the document]
Comment (BalzaniEdoardo) [reply docs/examples/plot_glm_pytree.py:106]: Agree

**Review by BalzaniEdoardo — CHANGES_REQUESTED** [review]
great work; I added a couple of commits. 
**First commit content:**
1. Add 1-line docstrings to tests.
2. Assert that tree_map returns a FeaturePytree
3. moved the --cov from the general config to the tox.ini call to pytest; the debugger behaves poorly if tests are configured with cov (doesn't stop at breakpoints)

**Second commit content:**
1. Added a test for asserting the equivalence of fitting a glm using pytrees or arrays

**Review by BalzaniEdoardo — APPROVED** [review]
good to go!

**Pull request merged**